alchemist/alchemist-fxui/src/main/java/it/unibo/alchemist/boundary/wormhole/implementation/AbstractWormhole2D.java

```diff
@@ -0,0 +1,387 @@
+package it.unibo.alchemist.boundary.wormhole.implementation;
```

Comment:
Yes, some files are used by fxui and swingui both. Namely:
- Hashes.kt
- AlchemistExtensions.kt
- Resources such as icons
- LocalizedResourceBundle.java
- The Wormhole interfaces and classes in it.unibo.alchemist.boundary.wormhole and its subpackages

As well as isolines and cognitive maps, when I add them to FXUI (after we're done reviewing but before merging?).

How should I handle these duplicated files?

Reply:
* Hashes can be moved in implementationbase
* `AlchemistExtensions` should not exist. Split it, if there are extensions for `Canvas`, create a `CanvasExtension`. See how it is done in cognitiveagents. I guess some functions are unused (e.g. the extensions functions on `Time`), so most of it can be leaned out
* The remainder common stuff should get its own subproject (alchemist-ui-tooling? Or propose a name, I'm open)

In any case, there must be ~zero duplication

Reply:
(I can't remember if I configured CPD to detect cut-paste... I should do it)